pkg/crc/machine/driver_darwin.go

```diff
@@ -35,3 +36,31 @@ func updateDriverConfig(host *host.Host, driver *machineVf.Driver) error {
 
 	return host.UpdateConfig(driverData)
 }
+
+func updateKernelArgs(vm *virtualMachine) error {
+	logging.Info("Updating kernel args...")
+	sshRunner, err := vm.SSHRunner()
+	if err != nil {
+		return err
+	}
+	defer sshRunner.Close()
+
+	stdout, stderr, err := sshRunner.RunPrivileged("Get kernel args", `-- sh -c 'rpm-ostree kargs'`)
```

Comment:
i think we need retry here, `crc stop` seems to fail sometime..
```
$ crc stop
INFO Stopping kubelet and all containers...
WARN Failed to stop all OpenShift containers.
Shutting down VM...
INFO Updating kernel args...
ERRO Failed to stop all containers: ssh command error:
command : sudo -- sh -c 'rpm-ostree kargs'
err     : ssh: handshake failed: EOF
 -
INFO Stopping the instance, this may take a few minutes...
Stopped the instance
```

Reply:
@anjannath This is ssh command failure, added a waitforConnectivity for it, do check once if that resolve it.

Reply:
Why is ssh down at this point, and why is it staying back up after waiting for it for a while? This seems to be papering over some other undiagnosed issue?

Reply:
> Why is ssh down at this point, and why is it staying back up after waiting for it for a while? This seems to be papering over some other undiagnosed issue?

Probably, i tested with this changes and that seemed to have fixed the issue then and did not investigate further.. but now i could not reproduce it with the new `4.11.1` bundle.. 

```
anjan at anath-mac in ~/work/crc-main on bebf89d5
$ crc stop --log-level debug
DEBU CRC version: 2.7.1+bebf89d5
DEBU OpenShift version: 4.11.1
DEBU Podman version: 4.1.1
DEBU Running 'crc stop'
DEBU Checking file: /Users/anjan/.crc/machines/crc/.crc-exist
DEBU Checking file: /Users/anjan/.crc/machines/crc/.crc-exist
DEBU Checking file: /Users/anjan/.crc/machines/crc/.crc-exist
INFO Stopping kubelet and all containers...
DEBU Using root access: Executing systemctl stop kubelet
DEBU Running SSH command: sudo systemctl stop kubelet
DEBU Using ssh private keys: [/Users/anjan/.crc/machines/crc/id_ecdsa /Users/anjan/.crc/cache/crc_vfkit_4.11.1_arm64/id_ecdsa_crc]
DEBU SSH command results: err: <nil>, output:
DEBU Using root access: stopping all containers
DEBU Running SSH command: sudo -- sh -c 'crictl stop $(crictl ps -q)'
DEBU SSH command results: err: <nil>, output: b19fa0ccd891d5aaa9919e9f15b6c7741a83260a56d994609f83a64beb86ab91
25f2648424cc02a16f79bd5e5f598160293e8bf37f85954e3137e2acf8494739
22fa3d56fb3d6fbd8c9e2ea080c47ae764cb69f5efd2f9c29fb38d436061a6e3
3907b24dcdc6e8522e7b6cbdf63545746ca12aee3c167beb5e37ce10dfd7ace7
4acacc57fcc38fecf4e56c79820c71276f253ae8aa6a7d7e518db9b8b66bdcff
b0bf6a5a92d21d500d74d07d799d2edb29a4d95bcb12f37420e4249a49911b5d
7f0ef1912702bf13da5e256dc30be329147178b975914f29161fbaca0336b3be
8ba4579fb1244284a4fb8359507221436e8c960d4bf5a62afb5599f795d72489
bc1292d908384b344076c237cbb09fd582fd02b0f1db874dd5ee72061f6a483f
7ccaf9b0aa5a98a0ee1f216375b324ff62192399557d1ec18d1558180d2e9625
c0d9d88492c280ae6b01c3962be5360d1c6282efa77b6497e8d3d84a7573f380
1a82e0dd3237e825aebf18a076ab0258233e81b028661b36cde41e8d8a26aae8
abacb936b7e14365827eeba9f88829455c451fd31c99ff4a3b98ff615d99f077
ae7e4e3827664b4a05787284f54ac2682fd2161f6420e0f80ac42177b16fd7de
54f3d1bb7d7a37dccde50bd499b5e6cbc052e8bb434742c5a205ba006efb0d69
28793959ff015b28324807361bcfc34ddb280976a266a8448a0af31869282dcc
1f7bbed6f60b3c0ad4ebfdb2fedd69380c25599cce8f2866af1f9503b411b820
14f79a7f8c3474b001b3fc0c99b9b3f480d2bfc82c5e74252dc0d008f7e6deb3
1ce33c850a70ea567cdc16acbd7100f5a31bcb83b3d468515d08091b9651fc31
8413bbb95d359ed458a5797f7d9bbcc8d2ec9c8489224b3ab4ce5a8705a04261
488c068b3f2f9a0982aa923f8b24a38af83dfbcf07efbcb0190bc82652b0a1eb
45aade99f9cb0310de2342d1f96cbf556f9ed87b23ebb07a2418a4a1dd6d2538
36198ba8ad70d093d98ced03b3763d6d1733de2833a9dcbcb03ce514858e571b
214b8b46257527ebcb464d3317ea8b8d15fc2c9b0c3256b1b4068457ff5ef1a5
bfd99711fd047591064065f8422e6eabdcc08cadd2ba6662086484b8add0f8d9
af0a7fa69b4eaac84e400fb244d4992ff01dff7d12bbc1bad8098731956450e3
f28761f1bd570c5f9852054fdd98ff2b2b2d32f7370f165cb2654371db9a37c7
f7599ac274442ac133d0a0f55de6a49ba33841b0cfe7eb45f0a9b75df5fb95d2
733067e39498c875faa44375ab9532c8d0d1e977e3dff30af3544c063841433e
f2ee135aaebaa0b9de92d7a15c69c45d067c1b259d243272d2432cff035b01ce
319239ed6e8cec31baf3f59cae004bd97f6686711d072bde820394b4219d6ad0
38df8cd85015797fd83402035cb9737b15fd4071613e69c79be6176ef84df87a
d1201110be84be3553a1552af059be8b2bd212db39a1d64b3b87aad53bd0118c
6d5cb0d288dcd9c7c6351460967d23f63cb73cbf6e31cb6c3b4203f60bc8de9b
cd14ffb42a9ca1f4567f846189306459995f43ed61be955b27b028ee5c150aa2
e59c56bdbb8b10558d55777f64f1b1dea51d2c76fc26d05105924a390dfc8451
f38c47b9605c01760d38cd3f07aeeb40aad53ba995f325307a8aa9404e3a530f
e8f7639039c755e700982d374d8be8c59185f6ec4c594fe23832314800ab32c0
bb54d89a1a2d056e087af12fc4ba83be27a36de014d7d56d10b81145c3c3309a
87cfe75a8b6ff420ea4c367e2c0e692cc6fa81b0205608207d766d1dab19e7dc
de0a50179aefc1d84af786a68e2f88ea3f61c8abe57f40339ddc871e5cf1affd
1ab5d801528c9402332008891f26270df2e5510330b7b4b7ce9050b3e6b7fd7b
1aa396a1e450b87d9f8a8b7e047d39268d0f798e0d07b63e22e3265b6f3ce541
a0711312be815469dc06165d04cad1d1118b035949c9e4bbadd76a059a03aa9a
b1476f5a48b8a58580716e3ee941a020a73ed49adfd2f8496d9460f414afd5e4
2cde74061cb3acfdf1e96f4e7497487914dc87d04fab928f55cbebc58c62d8c2
9aefcb56b43d62cdbe784576c69f24461e4f0b673e2db34e1e09d525d3cac968
44f55360c280f0677057de68e3750ed8aa55750ab84ae26e24f8f1a62d9d7515
09ca44ef990a855dfc5ff7dd43339ccd26e4df26d2574a1bef0dd42e1948be0c
c095ba5d679459f7e1af3d2210d803593ea5ccd8550e9948292538532ddc6ebb
8a49f4e01d072f4dd02ec8c36628b16b5724feb29a4d11a756f7c9774aeac565
ea6a67d4356a8b93f3912274ce6faf4af92954cb36d00374757a4875ead8edb5
c0e5aa55a533e3a6ab7464f199a3db373f6603d10da9765d7b6d1df2d1eddb26
2511c26f20c5cb443b0ebb9941c6bc022d5c1238b02f0dbeb8b99e8f6e2780df
7731111ca51972bb9cb4a098d465218e22c03e2da0ac46eb184c11488d5b9a5a
0816a7154849b721eb558c6a5417a31729f74008a0eb35da5c200a28a25b5722
6828da09754606e833725e5ab4762f930d9798d9fc90356bc458c69f591cb4af
43b715caecfd4e007669496ef7bc5eac19073c9965cffee98e2c7303d4c987cd
c5de858174457e280e23d3b00de8f787d6b112f076bc10d9f6f6e45d0d3d195e
ebda0e65c0d4dc1d4367a4a6565204505a8d67ba1d0af5791b0545f0ae40701d
697e3d8739bc85f47b4fc8bf34a33c02147593a83f017ec1225791f4f218e78f
00a833b18897f8d1680f8bf120b98b3c90196a4861ef1f9c90ea072c645e116c
91872bb6c0ae51eb950ce9188da99b60b9e0fd6d554aa6df5b4526647659c3f6
dc724f91ef532c3ba989192cf8fc2aebeb21f661bc8a8948f93e78bc122d8517
98fcc7e858045cbdfd975ba9e674fef37892a8988d4f8f6a605328d209b5f5ab
58868681a68788afcd32f66170aca6e2ee2005748c17d10c28bc90f8cc5cff6f
aec4ab8efd2e7c4085eb2274451ad8fce5099340eaef1e3208ab0698e83a80c2
c7cda20acf11d9fdeca2257795bd12855c5dfd0171417f6c6f01b41fcad0d49f
264efbf17ab7a55a2f0469458e66cc8a352f880ac40b06d9f1f901832edd322f
49302564f31d23cfc6ce0e2a22ae6f8230276638d4aff10a67ccc40fcde7d31c
9fe82ce4786dca7a5179dfa4f68ccb740138a81b0fe1049e850d708ecab702d2
3670f4ea4c68a920945c7d25e8ffe13221aedc1cf878cf2ee6f7b58c1983d168
2387dd6164938d4081df6e7d5582c76fe2b789cc85eaf309dabb9eaec6151ca8
b8d653e3332720a342c2e41cd5c6e3b413a72bd05d2fe93920a9caaccb0328dc
b5cce0cec9328992fa2905aac6e6288c9edb9053ef65fad1c35cd3d0e4f551e5
4df7bc53b0007dc0b28bfd41ee99cc7502229626559e0b548a0f3ffef5596cd7
5ae585c4632b9dc67d66295243ce370ced5dc81f07fd10c9179eef9b924fde36
82d3382cc861f2cc48af716857e23da2408e3fbefbcf891ebff001bc119ee122
920ba6cafd8e31b4313ce33f309b30bdee0aa0243bb2ba20b97bb9ab1bce7049
fa875a728aebdc5a1c360d59fff450c602134ef51f84c9f6f901fa60e335c7ad
b9a333a866660dcff99367659ea3c263bbea4364c16b8f17f4d7e7a10a11dbba
89ea41b6c96cf5d7671fd12a946f41ecaad999a7712fa2e849800086d472968b
bf3bcc85831c7738738e38ee95edf8c8a29e32c05ac56f552299d650eabdc297
dac829eddf290413b86375359c707e30cfe4a93496750932bf6bb661d13b47bf
840e774b054218d2a2a1f47fc0535ac5274b08e2605a7300889290d6446eb82c
d520fa816b797ec89d89099660d65a685daae3551cc69b1271824d61f51791dc
33fbd9cc9d1fac88c301cfbd9c15c2d28faa5c4e0e6e1c9cd028468a36ac2ad0
66ce2993e26bc10f7e13aff1b74d83380f1d3ca776056db8d0fff04157bfc8c6
6adbb517526e299efeceb0580ccf850d8498fe14a0b1b604197752dd9f057bef
c4e15a783d97e0ace67aaa7ec51c4a9302511271322ffa0e182c14168250bea1
dd4d06256caf5d64cb8eef0348de3d7a64b3f2094db6c612f10328006c03151a
1e048bc318b8fa4263dca22cac7f1eced1437010dfa97c23958d125e93e43109
INFO Updating kernel args...
DEBU Using root access: Get kernel args
DEBU Running SSH command: sudo -- sh -c 'rpm-ostree kargs'
DEBU Using ssh private keys: [/Users/anjan/.crc/machines/crc/id_ecdsa /Users/anjan/.crc/cache/crc_vfkit_4.11.1_arm64/id_ecdsa_crc]
DEBU SSH command results: err: <nil>, output: random.trust_cpu=on ignition.platform.id=qemu $ignition_firstboot ostree=/ostree/boot.1/rhcos/af091a0e76dc85593f8f5deeedce37851a992f53c8764aacac65b9e0f4dbffa9/0 root=UUID=38cffd8d-7f77-4d00-a96a-fd020e8d2af9 rw rootflags=prjquota boot=UUID=adad3366-de9e-461d-ae26-845f3de31cd8
DEBU Kernel args: random.trust_cpu=on ignition.platform.id=qemu $ignition_firstboot ostree=/ostree/boot.1/rhcos/af091a0e76dc85593f8f5deeedce37851a992f53c8764aacac65b9e0f4dbffa9/0 root=UUID=38cffd8d-7f77-4d00-a96a-fd020e8d2af9 rw rootflags=prjquota boot=UUID=adad3366-de9e-461d-ae26-845f3de31cd8
DEBU Current Kernel args: random.trust_cpu=on ignition.platform.id=qemu $ignition_firstboot ostree=/ostree/boot.1/rhcos/af091a0e76dc85593f8f5deeedce37851a992f53c8764aacac65b9e0f4dbffa9/0 root=UUID=38cffd8d-7f77-4d00-a96a-fd020e8d2af9 rw rootflags=prjquota boot=UUID=adad3366-de9e-461d-ae26-845f3de31cd8
DEBU /Users/anjan/.crc/machines/crc/crc.img is already 33285996544 bytes
INFO Stopping the instance, this may take a few minutes...
DEBU Stopping "crc"...
DEBU VM state: Stopped
DEBU retry loop: attempt 0
DEBU Machine "crc" was stopped.
Stopped the instance
```

Reply:
could not reproduce with even `4.10.0` bundle now (tested on M1)

Reply:
i could not reproduce it, i don't know what could have caused it before i think its fine to remove it, sorry about the false alarm!

Reply:
We can keep it if we want to err on the safe side, but I'd at least add a comment why this code is present.